src/Compilers/CSharp/Test/Symbol/Symbols/FunctionPointerTypeSymbolTests.cs

```diff
@@ -0,0 +1,762 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+#nullable enable
+
+using System.Linq;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class FunctionPointerTypeSymbolTests : CSharpTestBase
+    {
+        private void CommonVerifyFunctionPointer(FunctionPointerTypeSymbol symbol)
+        {
+            verifyPointerType(symbol);
+            verifySignature(symbol.Signature);
+            foreach (var param in symbol.Signature.Parameters)
+            {
+                verifyParameter(param, symbol.Signature);
+            }
+
+            static void verifyPointerType(FunctionPointerTypeSymbol symbol)
+            {
+                Assert.Equal(SymbolKind.FunctionPointer, symbol.Kind);
+                Assert.Equal(TypeKind.FunctionPointer, symbol.TypeKind);
+
+                Assert.False(symbol.IsReferenceType);
+                Assert.False(symbol.IsRefLikeType);
+                Assert.False(symbol.IsReadOnly);
+                Assert.False(symbol.IsStatic);
+                Assert.False(symbol.IsAbstract);
+                Assert.False(symbol.IsSealed);
+
+                Assert.True(symbol.IsValueType);
+
+                Assert.Null(symbol.ContainingSymbol);
+                Assert.Null(symbol.BaseTypeNoUseSiteDiagnostics);
+                Assert.Null(symbol.ObsoleteAttributeData);
+
+                Assert.Empty(symbol.Locations);
+                Assert.Empty(symbol.DeclaringSyntaxReferences);
+                Assert.Empty(symbol.GetMembers());
+                Assert.Empty(symbol.GetTypeMembers());
+                Assert.Empty(symbol.InterfacesNoUseSiteDiagnostics());
+            }
+
+            static void verifySignature(MethodSymbol symbol)
+            {
+                Assert.NotNull(symbol);
+
+                Assert.Equal(MethodKind.FunctionPointerSignature, symbol.MethodKind);
+                Assert.Equal(string.Empty, symbol.Name);
+                Assert.Equal(0, symbol.Arity);
+                Assert.Equal(default, symbol.ImplementationAttributes);
+                Assert.Equal(Accessibility.NotApplicable, symbol.DeclaredAccessibility);
+                Assert.Equal(FlowAnalysisAnnotations.None, symbol.ReturnTypeFlowAnalysisAnnotations);
+                Assert.Equal(FlowAnalysisAnnotations.None, symbol.FlowAnalysisAnnotations);
+
+                Assert.False(symbol.IsExtensionMethod);
+                Assert.False(symbol.HidesBaseMethodsByName);
+                Assert.False(symbol.IsStatic);
```

Comment:
consider asserting on symbol.RequiresInstanceReceiver as well.

Reply:
I don't think asserting that is of any particular use. The method won't be used. If anything, I would almost consider making it an `ExceptionUtilities.Unreachable`